lib/fluent/plugin/exception_detector.rb

```diff
@@ -59,6 +59,22 @@ def self.supported
            :java_start_exception),
       rule(:java_after_exception, /^[\r\n]*$/, :java_after_exception),
       rule([:java_after_exception, :java], /^[\t ]+(?:eval )?at /, :java),
+
+      # C# nested exception
+      rule([:java_after_exception, :java],
+            /^[\t ]+--- End of inner exception stack trace ---/,
+            :java),
+
+      # C# aggregate exception
+      rule([:java_after_exception, :java],
+            /^---> \(Inner Exception #[0-9]+\)/,
```

Comment:
These look sufficiently different from Java that they merit a separate set of rules for C# (possibly sharing some of the Java rules programmatically, e.g., by copying the rules and doing symbol substitution). @qingling128, WDYT?

Reply:
Agreed. At least we should separate the names (not having `java_after_exception` in a `C#` exception.

Reply:
The problem I'm having implementing this is there's not a good way of distinguishing between what is a C# exception and what is a Java exception when first transitioning to a `*_start_exception` state. Essentially both have to look for `Exception` in the text.

Because of this, if the `JAVA_RULES` are added first to `ALL_RULES` before `CSHARP_RULES`, then C# exceptions go off down the Java state machine path, and therefore never encounter C# specific rules like the ones I'm trying to implement. Likewise, if `CSHARP_RULES` were to appear before `JAVA_RULES` then Java exceptions get picked up by the C# state machine.